api/odigos/v1alpha1/instrumentedapplication_types.go

```diff
@@ -43,14 +43,25 @@ type OtherAgent struct {
 	Name string `json:"name,omitempty"`
 }
 
+type ProcessingState string
+
+const (
+	ProcessingStateFailed    ProcessingState = "Failed"
+	ProcessingStateSucceeded ProcessingState = "Succeeded"
+	ProcessingStateSkipped   ProcessingState = "Skipped"
+)
+
 // +kubebuilder:object:generate=true
 type RuntimeDetailsByContainer struct {
-	ContainerName  string                     `json:"containerName"`
-	Language       common.ProgrammingLanguage `json:"language"`
-	RuntimeVersion string                     `json:"runtimeVersion,omitempty"`
-	EnvVars        []EnvVar                   `json:"envVars,omitempty"`
-	OtherAgent     *OtherAgent                `json:"otherAgent,omitempty"`
-	LibCType       *common.LibCType           `json:"libCType,omitempty"`
+	ContainerName           string                     `json:"containerName"`
+	Language                common.ProgrammingLanguage `json:"language"`
+	RuntimeVersion          string                     `json:"runtimeVersion,omitempty"`
+	EnvVars                 []EnvVar                   `json:"envVars,omitempty"`
+	OtherAgent              *OtherAgent                `json:"otherAgent,omitempty"`
+	LibCType                *common.LibCType           `json:"libCType,omitempty"`
+	CriErrorMessage         *string                    `json:"criErrorMessage,omitempty"`
+	EnvFromContainerRuntime []EnvVar                   `json:"envFromContainerRuntime,omitempty"`
+	RuntimeUpdateState      *ProcessingState           `json:"runtimeUpdateState,omitempty"` // Tracks whether the new runtime detection process has been executed. If empty, the process has not been executed.
```

Comment:
Add a comment that this property is temporary and only used for the migration and showed be removed after some time